fuel/transformers/__init__.py

```diff
@@ -65,21 +73,30 @@ def get_epoch_iterator(self, **kwargs):
         return super(Transformer, self).get_epoch_iterator(**kwargs)
 
     def get_data(self, request=None):
```

Comment:
I like the `transform_example`, `transform_batch` setup. Could you write short docstrings for each of those (I don't think we really need a Parameters section, even).

What I kind of don't like is how `transform_any` is effectively a "hidden" part of the protocol because there's no base class method (though it is mentioned in the docs). Though it's kind of ugly, it might be better to throw catch a specific class of Exception (like `class NeedsStreamType(NotImplementedError): pass`), catch it in this method, and then dispatch one or the other of `transform_batch` or `transform_example`.

We could also write a really simple `StreamAgnosticTransformer` (or even just `AgnosticTransformer`) class that has an abstract `transform_any` and then `transform_batch` and `transform_example` implementations that call it, and direct users to subclass that if their transformation for batches and examples is the same.